incubator/types-core/src/main/java/io/aklivity/zilla/runtime/types/core/StringValidatorHandler.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.types.core;
+
+import org.agrona.DirectBuffer;
+
+import io.aklivity.zilla.runtime.engine.converter.function.ValueConsumer;
+import io.aklivity.zilla.runtime.engine.validator.ValidatorHandler;
+import io.aklivity.zilla.runtime.types.core.config.StringValidatorConfig;
+
+public class StringValidatorHandler implements ValidatorHandler
+{
+    private boolean initialize;
+    private int incompleteCharCount;
+    private int incompleteChar;
+
+    public StringValidatorHandler(StringValidatorConfig config)
+    {
+    }
+
+    @Override
+    public boolean validate(
+        int flags,
+        DirectBuffer data,
+        int index,
+        int length,
+        ValueConsumer next)
```

Comment:
Let's discuss what additional changes we need to make to the interface to support stateful validation of multi-byte chars across fragmentation boundaries.